pkg/clients/rds/rds.go

```diff
@@ -87,7 +91,7 @@ func (c *client) DescribeDBInstance(id string) (*DBInstance, error) {
 
 	response, err := c.rdsCli.DescribeDBInstances(request)
 	if err != nil {
-		return nil, err
+		return nil, errorConvert(err)
```

Comment:
Is purpose here just to return more descriptive error information?

Reply:
This is relate to： https://github.com/crossplane/provider-alibaba/blob/f7d0617918064fec5fbdc91eeb5d0b4d677913ba/pkg/controller/database/rdsinstance.go#L155 
if the Instance already remove from aliyun cloud, Observe should ignore the error

Reply:
What would you think about just modifying `IsErrorNotFound` to return true if either of these error types are set, rather than converting one error to the other?

Reply:
@hasheddan Already refactor the code as the suggestion, thanks.

Reply:
Thanks! This looks good to me :) I opened #47 to try to fix CI here

Reply:
Hi @hasheddan, i just add more test case for the code change and re trigger the CI.  all CI action is run and passed. Thanks.